diff --git a/Gemfile b/Gemfile
index 116c333..0cb7506 100644
--- a/Gemfile
+++ b/Gemfile
@@ -6,12 +6,10 @@ gem 'rails', '3.0.0'
 # gem 'rails', :git => 'git://github.com/rails/rails.git'
 
 gem 'sqlite3-ruby', :require => 'sqlite3'
-
 gem 'paperclip', :git => 'git://github.com/thoughtbot/paperclip.git'
-
 gem 'rio'
-
 gem 'acts_as_tree_rails3'
+gem 'faker'
 
 # Use unicorn as the web server
 # gem 'unicorn'
diff --git a/db/seeds.rb b/db/seeds.rb
index c6493a9..b35660c 100644
--- a/db/seeds.rb
+++ b/db/seeds.rb
@@ -6,21 +6,39 @@
 #   cities = City.create([{ :name => 'Chicago' }, { :name => 'Copenhagen' }])
 #   Mayor.create(:name => 'Daley', :city => cities.first)
 
-seen = []
-p = nil
+def add_categories
+  seen = []
+  p = nil
 
-rio('http://www.google.com/basepages/producttype/taxonomy.txt').read.split("\n").each do |l|
-  l.split(' > ').each do |c|
-   if seen.include?(c)
-      p = Category.where(['name = ?', c]).first
-   else
-      if p.nil?
-         p = Category.create!(:name => c)
+  rio('http://www.google.com/basepages/producttype/taxonomy.txt').read.split("\n").each do |l|
+    l.split(' > ').each do |c|
+      if seen.include?(c)
+        p = Category.where(['name = ?', c]).first
       else
-         p = p.children.create!(:name => c)
+        if p.nil?
+          p = Category.create!(:name => c)
+        else
+          p = p.children.create!(:name => c)
+        end
       end
-   end
-   seen << c
-  end  
-  p = nil
-end
\ No newline at end of file
+      seen << c
+    end  
+    p = nil
+  end
+end
+
+def add_companies
+  3000.times { add_company() }
+end
+
+def add_company
+  Company.create(:name => Faker::Company.name, :website => Faker::Internet.domain_name, :description => Faker::Lorem.sentences)
+end
+
+def random_category_id
+  n = Category.count
+  Category.select('id').limit(1).offset(rand(n)).id
+end
+
+add_categories
+add_companies

